### ISSUE TITLE
lower flatpak's xa.installed-size etc. to libostree

### ISSUE THREAD

**Issue by cgwalters** [ostreedev/ostree#1000]
Flatpak currently computes some data and adds it to the commit like [xa.installed-size](https://github.com/flatpak/flatpak/blob/eecdcbd1951d9c36a3c7d312da880e7d2eaa7a40/app/flatpak-builtins-build-export.c#L841).

I think we should lower this to libostree; it's useful for the host as well.  We do have some of this data also computed in static deltas, but the metadata is small, and nice to have everywhere.

There's a general argument that we should extend the metadata more; in https://github.com/ostreedev/ostree/pull/972 we added `ostree.ref-binding` and `ostree.collection-binding`.  

One specific metadata I'd propose adding as well is `ostree.n-directories` and `ostree.n-files`, so we can do a progress bar during checkout.

At the moment I'm thinking we could add an API like:

`void ostree_commit_add_standard_metadata_v0 (OstreeRepo *repo, OstreeMutableTree *root, const char *ref_binding, const char *collection_binding, GVariantBuilder *builder, GCancellable *cancellable, GError **error);`

And `ostree commit --metadata-v0`?

**Comment by cgwalters** [comment]
@alexlarsson What do you think?

**Comment by alexlarsson** [comment]
Yes, i think having this is useful in general. There is a risk that we bloat the summary file, but I think that is better solved by attacking the summary scalability in a more generic way (via e.g. partial updates).
